app/src/main/java/com/example/android/uamp/fragments/NowPlayingScreen.kt

```diff
@@ -0,0 +1,220 @@
+import coil.compose.AsyncImage
+import com.example.android.uamp.R
+import com.example.android.uamp.viewmodels.MainActivityViewModel
+import com.example.android.uamp.viewmodels.NowPlayingFragmentViewModel
+
+
+/**
+ * This particular instance of NowPlayingDescription keeps track of a mediaItem's state such that the
+ * correct mediaItemData is displayed i.e. the correct song being played, song's title, author, its
+ * respective album art, duration, and current position.
+ * Recomposes upon change in mediaItem's state.
+ *
+ * @param nowPlayingFragmentViewModel to reference NowPlayingFragmentViewModel functions
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ * @param navController to navigate to SettingsScreen
+ */
+@Composable
+fun NowPlayingDescription(
+    nowPlayingFragmentViewModel: NowPlayingFragmentViewModel,
+    mainActivityViewModel: MainActivityViewModel,
+    navController: NavController
+) {
+    val currentMediaItem by nowPlayingFragmentViewModel.mediaItem.observeAsState()
+    currentMediaItem?.let { mediaItem ->
+        NowPlayingDescription(
+            mediaItem,
+            nowPlayingFragmentViewModel,
+            mainActivityViewModel,
+            navController
+        )
+    }
+}
+
+/**
+ * This particular instance of NowPlayingDescription serves to showcase the current media item being played
+ * on the screen
+ *
+ * @param mediaItem current mediaItem being played
+ * @param nowPlayingFragmentViewModel to reference NowPlayingFragmentViewModel functions
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ * @param navController to navigate to SettingsScreen
+ */
+@OptIn(ExperimentalMaterialApi::class)
+@Composable
+private fun NowPlayingDescription(
+    mediaItem: MediaItem,
+    nowPlayingFragmentViewModel: NowPlayingFragmentViewModel,
+    mainActivityViewModel: MainActivityViewModel,
+    navController: NavController
+) {
+    if (mediaItem.equals(null)) {
+        Text(text = "Media Item is null")
+    } else {
+        Surface() {
+            NowPlaying(mediaItem, nowPlayingFragmentViewModel, mainActivityViewModel, navController)
+        }
+    }
+}
+
+/**
+ * NowPlaying describes the UI of the NowPlayingScreen which recomposes upon changes in the state of
+ * currentMediaItem's position, duration, and change in playback (play/pause).
+ *
+ * @param mediaItem current mediaItem being played
+ * @param nowPlayingFragmentViewModel to reference NowPlayingFragmentViewModel functions
+ * @param mainActivityViewModel to reference MainActivityViewModel functions
+ * @param navController to navigate to SettingsScreen
+ */
+@SuppressLint("PrivateResource")
+@Composable
+private fun NowPlaying(
+    mediaItem: MediaItem,
+    nowPlayingFragmentViewModel: NowPlayingFragmentViewModel,
+    mainActivityViewModel: MainActivityViewModel,
+    navController: NavController
+) {
+    val position: Long by nowPlayingFragmentViewModel.mediaPosition.observeAsState(0)
```

Comment:
This line itself is okay but the updates coming from NowPlayingFragmentViewModel.mediaPosition are inconsistent. It looks like the NowPlayingFragmentViewModel.checkPlaybackPosition() recursively calls itself roughly every second, but it can shift quite a bit. Worse, it never stops posting updates.

> checkPlaybackPosition(1000 - (currPosition % 1000))

That's the offending line, though I can't comment on it because it existed before this PR. The big problem is that the user can pause playback at say position 999ms. That means you get (1000 - (999 % 1000)) or (1000 - 999)... so every milisecond it creates a runnable to post to the main thread and trigger an update that calculates the exact same number that it previously calculated... forever 😬